src/main/scala/com/softwaremill/realworld/common/BaseEndpoints.scala

```diff
@@ -35,9 +36,11 @@ class BaseEndpoints(authService: AuthService):
 
   private def handleAuth(token: String): IO[ErrorInfo, UserSession] = {
     (for {
-      email <- authService.verifyJwt(token)
-    } yield UserSession(email)).logError.mapError {
+      userEmail <- authService.verifyJwt(token)
+      userId <- userIdByEmail(userEmail)
+    } yield UserSession(userId, userEmail)).logError.mapError {
       case e: Exceptions.Unauthorized => Unauthorized(e.message)
+      case e: Exceptions.NotFound     => NotFound(e.message)
```

Comment:
**suggestion:** The `NotFound` would be quite unexpected. It means that user got properly authenticated by email, but then we couldn't find their `userId`. This should never happen and I would propose returning an `InternalServerError` with e.message.

Reply:
But the code: 
```scala
userEmail <- authService.verifyJwt(token)
```
check only if token is correct and takes from this any email. This email can be invalid. So maybe better solution is making something like this: 
```scala
userEmail <- authService.verifyJwt(token) // return Unauthorized if wrong token
_ <- verifyEmail(userEmail) // return NotFound if token ok, but user not exists 
 userId <- userIdByEmail(userEmail) // create method to take id, but without Option if token and email ok
```